### PR TITLE
HADOOP-16453. Update how exceptions are handled in NetUtils.java

### PR DESCRIPTION
**Pull request by leosunli**
Signed-off-by: sunlisheng <sunlisheng@xiaomi.com>

### CODE REVIEW

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1079 | trunk passed |
| +1 | compile | 1044 | trunk passed |
| +1 | checkstyle | 41 | trunk passed |
| +1 | mvnsite | 70 | trunk passed |
| +1 | shadedclient | 725 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 55 | trunk passed |
| 0 | spotbugs | 112 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 111 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 47 | the patch passed |
| +1 | compile | 1030 | the patch passed |
| +1 | javac | 1030 | the patch passed |
| +1 | checkstyle | 38 | the patch passed |
| +1 | mvnsite | 66 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 629 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 54 | the patch passed |
| +1 | findbugs | 122 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 542 | hadoop-common in the patch passed. |
| +1 | asflicense | 44 | The patch does not generate ASF License warnings. |
| | | 5803 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1231 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux cf745b741009 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 3cc0ace |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/1/testReport/ |
| Max. process+thread count | 1449 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 84 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 2 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1224 | trunk passed |
| +1 | compile | 1182 | trunk passed |
| +1 | checkstyle | 53 | trunk passed |
| +1 | mvnsite | 87 | trunk passed |
| +1 | shadedclient | 897 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 62 | trunk passed |
| 0 | spotbugs | 135 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 133 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 55 | the patch passed |
| +1 | compile | 1040 | the patch passed |
| +1 | javac | 1040 | the patch passed |
| +1 | checkstyle | 48 | the patch passed |
| +1 | mvnsite | 85 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 744 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 67 | the patch passed |
| +1 | findbugs | 132 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 621 | hadoop-common in the patch passed. |
| +1 | asflicense | 46 | The patch does not generate ASF License warnings. |
| | | 6617 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/1231 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 0765c18eb250 4.15.0-54-generic #58-Ubuntu SMP Mon Jun 24 10:55:24 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 00b5a27 |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/2/testReport/ |
| Max. process+thread count | 1377 (vs. ulimit of 5500) |
| modules | C: hadoop-common-project/hadoop-common U: hadoop-common-project/hadoop-common |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 10 | https://github.com/apache/hadoop/pull/1231 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1231 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/3/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 13 | https://github.com/apache/hadoop/pull/1231 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1231 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/4/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

File: hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/net/TestNetUtils.java [hunk 1]
```diff
@@ -295,11 +293,8 @@ private  TestIOException(String cause){
       }
     }
     IOException e = new TestIOException();
-    IOException wrapped = verifyExceptionClass(e, IOException.class);
-    assertInException(wrapped, "Failed on local exception");
-    assertNotInException(wrapped, NetUtils.HADOOP_WIKI);
-    assertInException(wrapped, "Host Details ");
-    assertRemoteDetailsIncluded(wrapped);
+    IOException wrapped = verifyExceptionClass(e, TestIOException.class);
+    assertEquals(null, wrapped.getMessage());
```
Comment (steveloughran) [inline hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/net/TestNetUtils.java]: assertNull, with error text please

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 15 | https://github.com/apache/hadoop/pull/1231 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1231 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/5/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 10 | https://github.com/apache/hadoop/pull/1231 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hadoop/pull/1231 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-1231/7/console |
| versions | git=2.17.1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by steveloughran** [comment]
this is in; closing

**Pull request closed without merging**